From 64f9cf0034150491cc277d4230bf8d2c7bd9c441 Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:02 +0800
Subject: [PATCH 032/150] dt-bindings: w1: Add bindings for Phytium onewire
 controller

This patch documents the DT bindings for Phytium onewire controller.

Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: Ic39764d8670f5a7affd7057de5e82970153daa18
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 .../devicetree/bindings/w1/phytium,w1.yaml    | 33 +++++++++++++++++++
 MAINTAINERS                                   |  1 +
 2 files changed, 34 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/w1/phytium,w1.yaml

diff --git a/Documentation/devicetree/bindings/w1/phytium,w1.yaml b/Documentation/devicetree/bindings/w1/phytium,w1.yaml
new file mode 100644
index 000000000000..ea4b266cd292
--- /dev/null
+++ b/Documentation/devicetree/bindings/w1/phytium,w1.yaml
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/w1/phytium,w1.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium 1-wire bus master controller
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com.cn>
+
+properties:
+  compatible:
+    const: phytium,w1
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    maxItems: 1
+
+required:
+  - compatible
+  - reg
+  - interrupts
+
+examples:
+  - |
+    onewire0: onewire@2803f000 {
+      compatible = "phytium,w1";
+      reg = <0x0 0x2803f000 0x0 0x1000>;
+      interrupts = <GIC_SPI 166 IRQ_TYPE_LEVEL_HIGH>;
+    };
diff --git a/MAINTAINERS b/MAINTAINERS
index d4f59b28107c..68378f99ec68 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -2539,6 +2539,7 @@ F:	Documentation/devicetree/bindings/net/can/phytium,can.yaml
 F:	Documentation/devicetree/bindings/sound/phytium,hda.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,qspi-nor.yaml
 F:	Documentation/devicetree/bindings/spi/phytium,spi.yaml
+F:	Documentation/devicetree/bindings/w1/phytium,w1.yaml
 F:	arch/arm64/boot/dts/phytium/*
 F:	drivers/gpio/gpio-phytium*
 F:	drivers/i2c/busses/i2c-phytium-*
-- 
2.47.0

